### DISCUSSION TITLE
FontAwesome Pro icons with now

### DISCUSSION THREAD

**Discussion by vaklinzi** [vercel/vercel#3885] [General] [answered]
I am trying to deploy to `now` but I have the following error:

![image](https://user-images.githubusercontent.com/16841490/76092596-50af8700-5fc8-11ea-913e-cbd5d60ff182.png)

Before running the now command:
`$ now`
I run the following command for setting private token for the font awesome icons:
`$ npm config set "@fortawesome:registry" https://npm.fontawesome.com/ && npm config set "//npm.fontawesome.com/:_authToken" my_token`
Locally with `$ npm start` everything works fine but when I try with `$ now` or `$ now --prod`
I have the above error.
Maybe I need to set something like "preinstall" for the package.json or something similar so `now` run it before its setup.

**Comment by paulogdm** [comment, marked as answer]
Hello @vaklinzi. Thanks for posting in our GitHub community!

Check [this part](https://zeit.co/docs/v2/build-step#private-npm-modules-for-static-builds) of our documentation. Basically, you need to pass the private token to Now as an environment variable and the installation will succeed.

**EDIT**:
```
"build": {
  "env": {
    "NPM_TOKEN": "@my-now-secret"
  }
}
```
☝️ This is a "build-time environment variable".

**Reply by vaklinzi** [reply]
@paulogdm Thank you for the fast response and that you pointed me in the right direction!

I tried with this setup:

`{
  "version": 2,
  "public": false,
  "builds": [
    { "src": "package.json", "use": "@now/static-build" },
    { "src": "/api/server.js", "use": "@now/node-server" }
  ],
  "routes": [
    { "src": "/api/.*", "dest": "/api/server.js" },
    { "src": "^/static/(.*)", "dest": "/static/$1" },
    { "src": ".*", "dest": "/index.html" }
  ],
  "env": {
    "FONTAWESOME_TOKEN": "@fontawesome-token",
    "SECRET": "522533-fsdfds-3424324-fdsf"
  }
}`

`$ now secrets add FONTAWESOME_TOKEN 522533-fsdfds-3424324-fds`


but I still have the same error. Should the name something special?

**Reply by vaklinzi** [reply]
Also tried with:

`"env": {
    "NPM_TOKEN": "fortawesome-token"
  }`

`$ now secrets add  fortawesome-token 522533-fsdfds-3424324-fds`

![image](https://user-images.githubusercontent.com/16841490/76113974-ced25480-5fed-11ea-8035-08e7d8a99ebf.png)

after `$ now`

![image](https://user-images.githubusercontent.com/16841490/76114081-faedd580-5fed-11ea-8dc9-e2956ed83fdc.png)

I also tried with: `$ now  --build-env NPM_TOKEN=fortawesome-token`

**Comment by vaklinzi** [comment]
Found my problem:
It should be:
`"build": {
    "env": {
      "NPM_TOKEN": "@my-app"
    }
  }`
I was forgotten the "build". Thank you again @paulogdm

**Reply by paulogdm** [reply]
Glad you found it. I will edit my answer to clarify that.
